src/Symfony/Component/Lock/Lock.php

```diff
@@ -69,7 +77,7 @@ public function __destruct()
     public function acquire($blocking = false)
     {
         try {
-            if ($blocking) {
+            if ($blocking && $this->store instanceof BlockingStoreInterface && $this->store->supportsWaitAndSave()) {
```

Comment:
this is wrong (and a BC break). We need to throw a `NotSupportedException` here if we were requested to acquire a blocking lock, instead of silently turning it into a non-blocking one.